src/rc.c

```diff
@@ -178,7 +186,27 @@ read_rc(const char *file)
 		abs_file = NULL;
 	}
 
-	if (pfile) {
+	if (options.username == NULL) {
+		options.username = strdup(getenv("USER"));
+	}
+
+	if (options.pwprompt) {
+		prompt_password();
+	}
+
+#if HAVE_LIBSECRET
+	if (options.libsecret) {
+		if (!options.pwprompt) {
+			if (lookup_password() == 1) {
+				return(EXIT_FAILURE);
+			}
```

Comment:
Yeah, I'm thinking these are not fall backs, rather "or"s.  In my head the workflow would be to run `-p` the first time and then `libsecret` afterwards. Or you just use `.netrc` or `gpg`.  But not to prompt the user, since that is an CLI option.

Reply:
Re this and last comment - I think you're unambiguously right about **prompt vs .netrc/gpg** but I'm not so sure when it comes to **prompt vs. libsecret** as I think those two modes would typically go hand-in-hand with other applications.

But I *think* I get more or less what I want if `libsecret = yes` is specified in the config as then you can simply add `-p` if you need to specify what that password is.

What about a command line option as a compromise to enable this config to make it easier for a first-time user to play?